File: DataStructures/ExamPreparation-01/Problem-2-PitFortress/Java-Skeleton/PitFortress/src/main/models/Mine.java
package main.models;

import main.interfaces.IMine;

public class Mine implements IMine {

    private int id;
    private int delay;
    private int x;
    private Player player;
    private int damage;

    @Override
    public int getId() {
        return id;
    }

    @Override
    public int getDelay() {
        return delay;
    }

    public void setDelay(int value) {
        this.delay = value;
    }

    @Override
    public int getDamage() {
        return damage;
    }

    @Override
    public int getX() {
        return x;
    }

    @Override
    public Player getPlayer() {
        return player;
    }

    @Override
    public int compareTo(Mine o) {
        return 0;
    }
}

File: DataStructures/ExamPreparation-01/Problem-2-PitFortress/C#-Skeleton/PitFortressSkeleton/Classes/Mine.cs
namespace Classes
{
    using Interfaces;

    public class Mine : IMine
    {
        public int CompareTo(Mine other)
        {
            throw new System.NotImplementedException();
        }

        public int Id { get; private set; }

        public int Delay { get; set; }

        public int Damage { get; private set; }

        public int XCoordinate { get; private set; }

        public Player Player { get; private set; }
    }
}
